sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/compute/samples/CreateMultipleVirtualMachinesAndBatchQueryStatus.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.compute.samples;
+
+import com.azure.core.management.Region;
+import com.azure.resourcemanager.AzureResourceManager;
+import com.azure.resourcemanager.compute.models.CachingTypes;
+import com.azure.resourcemanager.compute.models.KnownLinuxVirtualMachineImage;
+import com.azure.resourcemanager.compute.models.VirtualMachineSizeTypes;
+import com.azure.resourcemanager.resourcegraph.ResourceGraphManager;
+import com.azure.resourcemanager.resourcegraph.models.QueryRequest;
+import com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerUtils;
+import com.azure.resourcemanager.samples.Utils;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Azure Compute sample for batch create virtual machines -
+ *   - Create multiple virtual machines with un-managed OS and data disks
+ *   - Use Azure Resource Graph to batch query the status (the number of provisioningState=Succeeded ?).
+ */
+public final class CreateMultipleVirtualMachinesAndBatchQueryStatus {
+
+    /**
+     * Main function which runs the actual sample.
+     *
+     * @param azureResourceManager instance of the azure client
+     * @param resourceGraphManager instance of the azure graph client
+     * @return true if sample runs successfully
+     */
+    public static boolean runSample(AzureResourceManager azureResourceManager, ResourceGraphManager resourceGraphManager) {
+        final int desiredVMCount = 6;
+        final Region region = Region.US_EAST;
+        String rgName = Utils.randomResourceName(azureResourceManager, "rg-", 15);
+        List<String> vmNameList = new ArrayList<>();
+        Long total = 0L;
```

Comment:
If total is Long, desiredVMCount should also be long.

Reply:
Fixed in the new version